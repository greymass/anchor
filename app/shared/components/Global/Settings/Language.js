// @flow
import React, { Component } from 'react';
import { Dropdown } from 'semantic-ui-react';
import { withTranslation } from 'react-i18next';

const languages = [
  { key: 'cn', value: 'zh-CN', flag: 'cn', text: '中文' },
  { key: 'en', value: 'en-US', flag: 'us', text: 'English' },
  { key: 'fr', value: 'fr-FR', flag: 'fr', text: 'Français' },
  { key: 'it', value: 'it-IT', flag: 'it', text: 'Italiano' },
  { key: 'ja', value: 'ja-JP', flag: 'jp', text: '日本語' },
  { key: 'kr', value: 'ko-KR', flag: 'kr', text: '한국어' },
  { key: 'ru', value: 'ru-RU', flag: 'ru', text: 'Русский' },
];

export class GlobalSettingsLanguage extends Component<Props> {
  onChange = (e, { value }) => {
    const { actions, i18n } = this.props;
    i18n.changeLanguage(value);
    actions.setSetting('lang', value);
  };

  render() {
    const {
      name,
      setLanguage,
      selection,
      t
    } = this.props;

    return (
      <Dropdown
        name={name}
        onChange={this.onChange}
        options={languages}
        placeholder={t('global_select_language')}
        selection={selection}
        value={setLanguage || 'en-US'}
      />
    );
  }
}

export default withTranslation('global')(GlobalSettingsLanguage);
