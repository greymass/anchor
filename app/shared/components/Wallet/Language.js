// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Dropdown, Flag, Menu } from 'semantic-ui-react';

const languages = [
  { key: 'en', value: 'en', flag: 'us', text: 'EN' },
  { key: 'fr', value: 'fr', flag: 'fr', text: 'FR' },
  { key: 'ja', value: 'ja', flag: 'jp', text: '日本語' },
  { key: 'kr', value: 'kr', flag: 'kr', text: '한글' }
];

class WalletLanguage extends Component<Props> {
  onChange = (e, { value }) => {
    const { actions, i18n } = this.props;
    i18n.changeLanguage(value);
    actions.setSetting('lang', value);
  }
  render() {
    const {
      settings
    } = this.props;
    return (
      <Menu.Item
        key="language"
        inverted="true"
      >
        <Dropdown
          defaultValue={settings.lang}
          fluid
          onChange={this.onChange}
          options={languages}
        />
      </Menu.Item>
    );
  }
}

export default translate('wallet')(WalletLanguage);
