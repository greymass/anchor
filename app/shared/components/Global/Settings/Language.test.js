import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalSettingsLanguage } from './Language';

const mockProps = {
  name: 'transfer filter',
  selection: false,
  setLanguage: 'en-USapp/shared/components/Global/Settings/FilterSpamTransfers.test.js',
  t: text => text,
};

describe('GlobalSettingsLanguage', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalSettingsLanguage {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
