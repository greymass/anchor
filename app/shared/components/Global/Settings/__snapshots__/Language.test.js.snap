// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSettingsLanguage renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  icon="dropdown"
  minCharacters={1}
  name="transfer filter"
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "flag": "cn",
        "key": "cn",
        "text": "中文",
        "value": "zh-CN",
      },
      Object {
        "flag": "us",
        "key": "en",
        "text": "English",
        "value": "en-US",
      },
      Object {
        "flag": "fr",
        "key": "fr",
        "text": "Français",
        "value": "fr-FR",
      },
      Object {
        "flag": "it",
        "key": "it",
        "text": "Italiano",
        "value": "it-IT",
      },
      Object {
        "flag": "jp",
        "key": "ja",
        "text": "日本語",
        "value": "ja-JP",
      },
      Object {
        "flag": "kr",
        "key": "kr",
        "text": "한국어",
        "value": "ko-KR",
      },
      Object {
        "flag": "ru",
        "key": "ru",
        "text": "Русский",
        "value": "ru-RU",
      },
    ]
  }
  placeholder="global_select_language"
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={false}
  value="en-USapp/shared/components/Global/Settings/FilterSpamTransfers.test.js"
  wrapSelection={true}
/>
`;
